jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/skaffold/FilesMojo.java

```diff
@@ -0,0 +1,153 @@
+    Preconditions.checkNotNull(session);
+    Preconditions.checkNotNull(extraDirectory);
+    Preconditions.checkNotNull(projectDependenciesResolver);
+
+    // print out pom configuration files
+    System.out.println(project.getFile());
+    if ("pom".equals(project.getPackaging())) {
+      // done if <packaging>pom</packaging>
+      return;
+    }
+
+    // print out sources directory (resolved by maven to be an absolute path)
+    System.out.println(project.getBuild().getSourceDirectory());
+
+    // print out resources directory (resolved by maven to be an absolute path)
+    ImmutableSet.copyOf(project.getBuild().getResources())
+        .stream()
+        .map(FileSet::getDirectory)
+        .forEach(System.out::println);
+
+    // this seems weird, but we will only print out the jib "extraFiles" directory on projects where
+    // the plugin is explicitly configured (even though _skaffold-files is a jib-maven-plugin goal
+    // and is expected to run on all projects irrespective of their configuring of the jib plugin).
+    if (project.getPlugin(MavenProjectProperties.PLUGIN_KEY) != null) {
+      // print out extra directory
+      System.out.println(extraDirectory.getAbsoluteFile().toPath());
+    }
+
+    // Grab non-project SNAPSHOT dependencies for this project
+    // TODO: this whole sections relies on internal maven API, it could break. We need to explore
+    // TODO: better ways to resolve dependencies using the public maven API.
+    Set<String> projectArtifacts =
+        projects
+            .stream()
+            .map(MavenProject::getArtifact)
+            .map(Artifact::toString)
+            .collect(Collectors.toSet());
+
+    DependencyFilter ignoreProjectDependenciesFilter =
+        (node, parents) -> {
+          if (node == null || node.getDependency() == null) {
+            // if nothing, then ignore
+            return false;
+          }
+          if (projectArtifacts.contains(node.getArtifact().toString())) {
+            // ignore project dependency artifacts, ignore
+            return false;
+          }
+          // we only want compile/runtime deps
+          return Artifact.SCOPE_COMPILE_PLUS_RUNTIME.contains(node.getDependency().getScope());
+        };
+
+    try {
+      DependencyResolutionResult resolutionResult =
+          projectDependenciesResolver.resolve(
+              new DefaultDependencyResolutionRequest(project, session.getRepositorySession())
+                  .setResolutionFilter(ignoreProjectDependenciesFilter));
+      resolutionResult.getDependencies()
+          .stream()
+          .map(org.eclipse.aether.graph.Dependency::getArtifact)
+          .filter(org.eclipse.aether.artifact.Artifact::isSnapshot)
+          .map(org.eclipse.aether.artifact.Artifact::getFile)
+          .forEach(System.out::println);
+
+    } catch (DependencyResolutionException ex) {
+      throw new MojoExecutionException("Failed to resolve dependencies", ex);
```

Comment:
Is this really the right thing to do?  If I specify the wrong version, this is what I see:

```
$ mvn -q jib:_skaffold-files
/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/pom.xml
/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/src/main/java
/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/src/main/resources
/Users/bsd/Manumitting/Projects/GooglePluginEclipse/jibplayground/maven/app/src/main/jib
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:0.9.11-SNAPSHOT:_skaffold-files (default-cli) on project first: Failed to resolve dependencies: Could not resolve dependencies for project ca.mt.jib.examples:first:jar:0.0.1-SNAPSHOT: Failure to find com.google.cloud.tools:appengine-plugins-core:jar:0.6.0-SNAPSHOT in http://nexus.mt.ca:8081/nexus/content/groups/public was cached in the local repository, resolution will not be reattempted until the update interval of nexus has elapsed or updates are forced -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```
(Note: the Maven error logging goes to stdout too.)

Should Skaffold ignore the results with a non-zero exit code and retry after some timeout?

Reply:
I think the error result should actually result in an error. I'm also wondering if I should change this to just calculate the correct filename from the artifact coordinates, instead of explicitly resolving (downloading) everything.

Reply:
Oh yeah, we shouldn't be "stopping" skaffold during a dev run. But we have to surface those errors to the users.

That error is also a little weird, I believe us using ProjectDependencyResolver in a weird way is providing a result that doesn't make full sense.